source/vir/src/headers.rs

```diff
@@ -238,3 +240,112 @@ impl Header {
         Arc::new(invs)
     }
 }
+
+fn make_trait_decl(method: &Function, spec_method: &Function) -> Result<Function, VirErr> {
+    let FunctionX {
+        name: _,
+        kind: _,
+        visibility: _,
+        mode: _,
+        fuel,
+        typ_bounds,
+        params,
+        ret,
+        require,
+        ensure,
+        decrease,
+        decrease_when,
+        decrease_by,
+        broadcast_forall: _,
+        mask_spec,
+        is_const: _,
+        publish: _,
+        attrs: _,
+        body: _,
+        extra_dependencies,
+    } = spec_method.x.clone();
+    let mut methodx = method.x.clone();
+    if methodx.typ_bounds.len() != typ_bounds.len() {
+        return err_str(
+            &spec_method.span,
+            "method specification has different number of type parameters from method",
+        );
+    }
+    for (b1, b2) in methodx.typ_bounds.iter().zip(typ_bounds.iter()) {
+        let ((x1, g1), (x2, g2)) = (b1, b2);
+        match (&**g1, &**g2) {
+            (GenericBoundX::Traits(ps1), GenericBoundX::Traits(ps2)) => {
+                if x1 != x2 || ps1 != ps2 {
+                    return err_str(
+                        &spec_method.span,
+                        "method specification different type parameters from method",
```

Comment:
i think this should say 'type bounds' not 'type parameters' right?

Also can we add test cases for both inconsistent type bounds & inconsistent type parameters?

Reply:
It's hard to make a real test case yet because trait method type parameters aren't actually supported, but I can add a test case that will work in the future.